ssl/t1_enc.c

```diff
@@ -481,18 +447,23 @@ int tls1_change_cipher_state(SSL *s, int which)
         goto err;
     }
 
-# ifndef __FreeBSD__
     if (which & SSL3_CC_WRITE)
         rl_sequence = RECORD_LAYER_get_write_sequence(&s->rlayer);
     else
         rl_sequence = RECORD_LAYER_get_read_sequence(&s->rlayer);
 
+# ifdef __FreeBSD__
+    if (!ktls_configure_crypto(s, c, dd, rl_sequence, &crypto_info, &rec_seq,
+                               iv, key, ms, *mac_secret_size))
+        goto skip_ktls;
+# else
     if (!ktls_configure_crypto(c, s->version, dd, rl_sequence, &crypto_info,
```

Comment:
We could perhaps remove this #ifdef here and in tls13_enc.c if we were to use the arguments from the FreeBSD call (it is a superset) for Linux as well.  Linux could just ignore mac secrets, etc.

Reply:
Again - yes, that would be good.

Reply:
Done for both callbacks.